analyzers/src/SonarAnalyzer.Common/SyntaxTrackers/PropertyAccessTracker.cs

```diff
@@ -20,91 +20,50 @@
-                    }
-                });
-
-            void TrackMemberAccess(SyntaxNodeAnalysisContext c)
-            {
-                if (IsTrackedProperty(c.Node, c.SemanticModel))
-                {
-                    c.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, c.Node.GetLocation(), diagnosticMessageArgs));
-                }
-            }
-
-            bool IsTrackedProperty(SyntaxNode expression, SemanticModel semanticModel)
-            {
-                // We register for both MemberAccess and IdentifierName and we want to
-                // avoid raising two times for the same identifier.
-                if (IsIdentifierWithinMemberAccess(expression))
-                {
-                    return false;
-                }
-
-                var propertyName = GetPropertyName(expression);
-                if (propertyName == null)
-                {
-                    return false;
-                }
-
-                var conditionContext = new PropertyAccessContext(expression, propertyName, semanticModel);
-                return conditions.All(c => c(conditionContext));
-            }
-        }
+        private bool CaseInsensitiveComparison { get; }
 
-        #region Syntax-level standard conditions
+        protected PropertyAccessTracker(IAnalyzerConfiguration analyzerConfiguration, DiagnosticDescriptor rule, bool caseInsensitiveComparison = false) : base(analyzerConfiguration, rule) =>
+           CaseInsensitiveComparison = caseInsensitiveComparison;
 
         public PropertyAccessCondition MatchProperty(params MemberDescriptor[] properties) =>
             context =>
                 MemberDescriptor.MatchesAny(context.PropertyName, context.PropertySymbol, false, CaseInsensitiveComparison, properties);
 
-        public abstract PropertyAccessCondition MatchGetter();
-
-        public abstract PropertyAccessCondition MatchSetter();
-
-        public abstract PropertyAccessCondition AssignedValueIsConstant();
-
-        #endregion
+        protected override BaseContext IsTracked(SyntaxNode expression, SemanticModel semanticModel, PropertyAccessCondition[] conditions, out Location location)
+        {
+            location = Location.None;
 
-        #region Symbol-level standard conditions
+            // We register for both MemberAccess and IdentifierName and we want to
+            // avoid raising two times for the same identifier.
+            if (IsIdentifierWithinMemberAccess(expression))
+            {
+                return null;
+            }
 
-        // Add any common symbol-level checks here...
+            var propertyName = GetPropertyName(expression);
+            if (propertyName == null)
+            {
+                return null;
+            }
 
-        #endregion
+            var context = new PropertyAccessContext(expression, propertyName, semanticModel);
+            location = expression.GetLocation();
+            return conditions.All(c => c(context)) ? context : null;
```

Comment:
After adressing other comments, this could be something like
```
return GetPropertyName(expression) is {} propertyName
    ? CreateContext(...)
    : null;